### PR TITLE
add i/o write stream. closes #29

### LINKED ISSUE #29: Add optional I/O write stream for sync'd logs
Per https://github.com/timberio/library-specification/blob/master/SPECIFICATION.MD#332-io-device-formatter

For example -

This could be a function that takes a stream:

```
// Only the Node.js lib...
timber.writeStream(someStream)
```

... which will then be piped to _at time of syncing_.

Or perhaps the API could be more stream-like on the `timber` instance directly:

```
timber.pipe(writeStream)
```

... which would then allow all of the usual stream functions (including - in an upcoming/separate issue - the notion of reading from a stream as a form of log ingestion.)

@aks- open to your thoughts.

### PR DESCRIPTION
**Pull request by aks-**

### CODE REVIEW

**Comment by leebenson** [comment]
@aks- - one thing I noticed is that you've added stream functionality to `@timberio/core` instead of `@timberio/node`.

Was there a specific design decision that led to it being included in core?

I ask because the purpose of this functionality is to take advantage of Node's Stream API specifically. It's unlikely to have any benefit to the browser, so will likely just bloat the UMD bundle being in the Base class.

The main reasons for adding streams is to add compatibility to:

1. Existing logging solutions that rely on streams. The Timer instance should be _writeable_ as well as _readable_ (duplex), so that logs coming from third-party streams can be transformed into the common `ITimberLog` format before being synced. 

2. Interop with low-level Node.js I/O, so logs can be saved to disk, sent to a network stream, etc.

In summary - it should be somewhat compatible with the likes of [this cheatsheet](https://devhints.io/nodejs-stream).

I probably didn't explain the use-case well enough by suggesting a`.writeStream()` method.

Let me spend a bit of time thinking about a more precise spec for streams, and I'll come back to you with details. Thanks.

**Comment by aks-** [comment]
hey @leebenson , thanks. i missed that i could overload `_batch` hence didn't see any other way to write transformed logs to writable streams. i'm now bit confused about the impl of the streams. as per this https://github.com/timberio/timber-js/issues/29#issuecomment-446061700 i thought we were only writing logs to a writable stream... but looks like we want an interface where timber is more of a transform stream? where at one end user would write to the stream `rs.pipe(timer).pipe(ws); rs.push('hey');` and we would transform that message and write it to writable stream?
